service/src/integrationTest/kotlin/fi/espoo/evaka/invoicing/service/FeeDecisionGeneratorIntegrationTest.kt

```diff
@@ -2507,6 +2508,52 @@ class FeeDecisionGeneratorIntegrationTest : FullApplicationTest(resetDbBeforeEac
             }
         }
     }
+    @Test
+    fun `a new fee decision is not generated if incomes change from NOT_AVAILABLE to INCOMPLETE`() {
+        val period = DateRange(LocalDate.of(2019, 1, 1), LocalDate.of(2019, 12, 31))
+        val incomePeriod = period.copy(end = null)
+        insertFamilyRelations(testAdult_1.id, listOf(testChild_1.id), period)
+        insertPlacement(testChild_1.id, period, DAYCARE, testDaycare.id)
+        insertIncome(testAdult_1.id, 310200, incomePeriod, IncomeEffect.NOT_AVAILABLE)
+        insertIncome(testChild_1.id, 310200, incomePeriod, IncomeEffect.NOT_AVAILABLE)
+
+        db.transaction {
+            generator.generateNewDecisionsForAdult(
+                it,
+                RealEvakaClock(),
+                testAdult_1.id,
+                period.start
+            )
+        }
+
+        val decisions = getAllFeeDecisions()
+        assertEquals(1, decisions.size)
+
+        feeDecisionController.confirmDrafts(
+            dbInstance(),
+            AuthenticatedUser.Employee(testDecisionMaker_2.id, setOf(UserRole.ADMIN)),
+            RealEvakaClock(),
+            listOf(decisions.get(0).id),
+            null
+        )
+
+        asyncJobRunner.runPendingJobsSync(RealEvakaClock())
+
+        db.transaction { it.createUpdate("UPDATE income SET effect = 'INCOMPLETE'").execute() }
+
+        db.transaction {
+            generator.generateNewDecisionsForAdult(
+                it,
+                RealEvakaClock(),
+                testAdult_1.id,
+                incomePeriod.start
+            )
+        }
+
+        // No new DRAFT is generated because the only diff was head of family income type change
+        // NOT_AVAILABLE -> INCOMPLETE
+        assertEquals(1, getAllFeeDecisions().size)
+    }
```

Comment:
I would like to see more tests. The case where a child has incomes is very rare, so this is not a very realistic test.

More interesting tests would be:
* the income of a single parent changes
* the income of one adult changes in a two-adult family